llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -1308,9 +1390,9 @@ void Cost::RateRegister(const Formula &F, const SCEV *Reg,
 
       // If the step size matches the base offset, we could use pre-indexed
       // addressing.
-      if (AMK == TTI::AMK_PreIndexed) {
+      if (AMK == TTI::AMK_PreIndexed && !F.BaseOffset.isScalable()) {
```

Comment:
There's no 'isFixed()' method on FixedOrScalableQuantity, so I just used the inverse of isScalable(). They're logically equivalent, though there's semantic reinforcement from separating them.

But I do wonder if we want an uninitialized state for scalable-ness, and allow that to be freely used with either fixed or scalable offsets. The result would then be 'locked in' to one or the other.

Reply:
I'm suggesting to add the new function because they're not logically equivalent. It's just the "zero" support is incomplete and my worry is that once that's fixed, code like this will silently break.

Reply:
Immediate::isZero() is the same function you've referenced -- it ignores the Scalable flag and just returns Quantity. isScalable() returns Scalable, and a new isFixed() would return '!Scalable'.

Do we want isScalable() to mean 'Scalable || Quantity == isZero()' and change the other FixedOrScalableQuantity types too?